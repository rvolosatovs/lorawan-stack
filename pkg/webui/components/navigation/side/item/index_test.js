// Copyright Â© 2019 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import getSideNavigationItemDriver from './index_driver'

describe('SideNavigationItem', function() {
  let driver = null
  let onExpandSpy = null

  beforeEach(function() {
    driver = getSideNavigationItemDriver()
    onExpandSpy = jest.fn()
  })

  describe('is flat', function() {
    beforeEach(function() {
      driver.when.created({
        title: 'test-title',
        path: '/test-title',
        depth: 0,
        isCollapsable: false,
        isMinimized: false,
        isExpanded: false,
        isActive: false,
        items: [],
        onExpand: onExpandSpy,
      })
    })

    it('should match snapshot', function() {
      expect(driver.component).toMatchSnapshot()
    })
  })

  describe('is collapsable', function() {
    beforeEach(function() {
      driver.when.created({
        title: 'test-title',
        depth: 0,
        isCollapsable: true,
        isMinimized: false,
        isExpanded: true,
        isActive: false,
        onExpand: onExpandSpy,
        items: [
          {
            title: 'nested-title',
            path: '/nested-title',
          },
          {
            title: 'nested-title2',
            path: '/nested-title2',
          },
        ],
      })
    })

    it('should match snapshot', function() {
      expect(driver.component).toMatchSnapshot()
    })

    describe('the user selects the item', function() {
      beforeEach(function() {
        driver.when.itemSelected()
      })

      it('should call `onExpand` only once', function() {
        expect(onExpandSpy.mock.calls).toHaveLength(1)
      })
    })
  })
})
