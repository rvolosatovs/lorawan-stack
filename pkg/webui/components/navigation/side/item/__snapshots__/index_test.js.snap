// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigationItem is collapsable should match snapshot 1`] = `
<li
  className="item"
>
  <CollapsableItem
    depth={0}
    icon={null}
    isActive={false}
    isExpanded={true}
    isMinimized={false}
    items={
      Array [
        Object {
          "path": "/nested-title",
          "title": "nested-title",
        },
        Object {
          "path": "/nested-title2",
          "title": "nested-title2",
        },
      ]
    }
    onClick={[Function]}
    onExpand={[MockFunction]}
    title="test-title"
  />
</li>
`;

exports[`SideNavigationItem is flat should match snapshot 1`] = `
<li
  className="item"
>
  <LinkItem
    depth={0}
    exact={true}
    icon={null}
    onExpand={[Function]}
    path="/test-title"
    title="test-title"
  />
</li>
`;
